### PR TITLE
Switch from shed to ruff

### PR DESCRIPTION
**Pull request by auscompgeek**
https://github.com/charliermarsh/ruff

## Checklist

- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [ ] I have not broken the cheatsheet

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[tool.black]
+target-version = ["py39"]
+
+[tool.ruff]
+select = ["E", "F", "C4", "I001", "UP", "SIM"]
```
Comment (auscompgeek) [inline pyproject.toml:5]: This enables the closest set of rules that `shed` fixed without being overly specific.

- `F`: pyflakes/autoflake
- [`C4`: flake8-comprehensions](https://beta.ruff.rs/docs/rules/#flake8-comprehensions-c4) - shed has some custom autofixers
- `I001`: isort
- `UP`: pyupgrade
- [`SIM`: flake8-simplify](https://beta.ruff.rs/docs/rules/#flake8-simplify-sim) - I think shed has a couple of these autofixers

**Review by pokey — APPROVED** [review]
Looks great to me! @phillco @AndreasArvidsson ?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ignore = ["E501", "SIM105"]
+target-version = "py39"
+extend-exclude = ["vendor"]
+
+[tool.ruff.pyupgrade]
+keep-runtime-typing = true
```
Comment (pokey) [inline pyproject.toml:11]: Does this fix that issue where Talon choked on the newer type annotations?

Comment (auscompgeek) [reply pyproject.toml:11]: I _think_ so, but not 100% certain. Here's the docs for this setting: https://beta.ruff.rs/docs/settings/#keep-runtime-typing

Comment (pokey) [reply pyproject.toml:11]: Huh. Strange that it says:

> Note that this setting is only applicable when the target Python version is below 3.9 and 3.10 respectively

So I'm not sure this rule even applies? Maybe instead we should just disable `UP006`? That corresponds to [PEP 585](https://peps.python.org/pep-0585/), which I'm pretty sure is the troublemaker.

Comment (pokey) [reply pyproject.toml:11]: Huh. Strange that it says:

> Note that this setting is only applicable when the target Python version is below 3.9 and 3.10 respectively

So I'm not sure this rule even applies? Maybe instead we should just disable `UP006`? That corresponds to [PEP 585](https://peps.python.org/pep-0585/), which I'm pretty sure is the troublemaker.

Comment (pokey) [reply pyproject.toml]: Ok I did some local testing and looks like these rules would still have caused https://github.com/knausj85/knausj_talon/issues/922.  Disabling `UP035` did the trick, though.  I pushed updated config; lmk if it looks ok and we can merge away

Thanks for this one!  So much faster.  Now prettier is the slow horse 😄

**Review by pokey — APPROVED** [review]

**Comment by auscompgeek** [comment]
Yep, I was planning to enable more rules as a follow-up. Was just aiming to match shed to start.

**Pull request merged**